test/test_dr.py

```diff
@@ -0,0 +1,58 @@
+import ot
+import numpy as np
+import pytest
+
+try:  # test if cudamat installed
+    import ot.dr
+    nogo = False
+except ImportError:
+    nogo = True
+
+
+@pytest.mark.skipif(nogo, reason="Missing modules (autograd or pymanopt)")
+def test_fda():
+
+    n = 90  # nb samples in source and target datasets
+    np.random.seed(0)
+
+    # generate circle dataset
+    xs, ys = ot.datasets.get_data_classif('gaussrot', n)
+
+    nbnoise = 8
+
+    xs = np.hstack((xs, np.random.randn(n, nbnoise)))
+
+    p = 1
+
+    Pfda, projfda = ot.dr.fda(xs, ys, p)
+
+    projfda(xs)
+
+    assert np.allclose(np.sum(Pfda**2, 0), np.ones(p))
+
+
+@pytest.mark.skipif(nogo, reason="Missing modules (autograd or pymanopt)")
+def test_wda():
+
+    n = 100  # nb samples in source and target datasets
+    nz = 0.2
+    np.random.seed(0)
+
+    # generate circle dataset
+    t = np.random.rand(n) * 2 * np.pi
+    ys = np.floor((np.arange(n) * 1.0 / n * 3)) + 1
+    xs = np.concatenate(
+        (np.cos(t).reshape((-1, 1)), np.sin(t).reshape((-1, 1))), 1)
+    xs = xs * ys.reshape(-1, 1) + nz * np.random.randn(n, 2)
+
+    nbnoise = 8
+
+    xs = np.hstack((xs, np.random.randn(n, nbnoise)))
+
+    p = 2
+
+    Pwda, projwda = ot.dr.wda(xs, ys, p, maxiter=10)
```

Comment:
if you step back a little guessing for a new user what `ot.dr.wda` means it's a pretty hard challenge :)

maybe you can leaving with a bit more typing and use less jardon/acronyms